dask/dataframe/tests/test_dataframe.py

```diff
@@ -1992,14 +1992,42 @@ def test_repartition_freq_errors():
 
 
 def test_repartition_freq_month():
-    ts = pd.date_range("2015-01-01 00:00", " 2015-05-01 23:50", freq="10min")
+    ts = pd.date_range("2015-01-01 00:00", "2015-05-01 23:50", freq="10min")
     df = pd.DataFrame(
         np.random.randint(0, 100, size=(len(ts), 4)), columns=list("ABCD"), index=ts
     )
-    ddf = dd.from_pandas(df, npartitions=1).repartition(freq="1M")
+    ddf = dd.from_pandas(df, npartitions=1).repartition(freq="MS")
 
     assert_eq(df, ddf)
-    assert 2 < ddf.npartitions <= 6
+
+    assert ddf.divisions == (
+        pd.Timestamp("2015-1-1 00:00:00", freq="MS"),
+        pd.Timestamp("2015-2-1 00:00:00", freq="MS"),
+        pd.Timestamp("2015-3-1 00:00:00", freq="MS"),
+        pd.Timestamp("2015-4-1 00:00:00", freq="MS"),
+        pd.Timestamp("2015-5-1 00:00:00", freq="MS"),
+        pd.Timestamp("2015-5-1 23:50:00", freq="10T"),
```

Comment:
It seems strange that not all of the divisions would have the same freq when going through `repartition_freq`. Probably minor since the divisions aren't part of a collection with an overall freq like a `DatetimeIndex`, but do you know why this is?

Reply:
It's indeed ugly, but it's already the case at the moment because the new divisions are built up from the old divisions (which can have any `freq`) and a `pd.date_range` with the `freq` given as argument.

I think best would be to remove the `freq` property from each timestamp (since it only causes confusion).

Reply:
I agree with this assessment. freq on divisions doesn't really make sense.